server/py/framework/utils/time_window_tracker.py

```diff
@@ -141,3 +130,12 @@ async def run_with_time_window_tracker(
             cycle_tracker.update_window,
             now,
         )
+        ensure_window_update = False
+    finally:
+        await run_in_threadpool(framework.db.close_session, db_session)
+        if ensure_window_update:
+            await run_in_threadpool(
+                framework.db.session.run_function_with_new_db_session,
+                cycle_tracker.update_window,
+                now,
+            )
```

Comment:
I'd also add a comment here since its not trivially obvious why you are closing the session and immediately opening a new one